### PR TITLE
Feature/room-list

### PR DESCRIPTION
**Pull request by DiveshPatel**
This PR references issue #29 where a user can use the search bar to search for all available rooms and automatically show the location of the room on the map. Flutter format was already run

### CODE REVIEW

File: lib/widgets/search/main_search_bar.dart [hunk 1]
```diff
@@ -1,9 +1,13 @@
+import 'package:concordi_around/data/building_singleton.dart';
 import 'package:concordi_around/models/coordinate.dart';
 import 'package:concordi_around/widgets/search/search_menu.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter/rendering.dart';
 import 'package:google_maps_flutter/google_maps_flutter.dart';
 
+import '../../models/building.dart';
+import '../../models/coordinate.dart';
+
```
Comment (MrJCipherButtles) [inline lib/widgets/search/main_search_bar.dart]: Watch for duplicated imports

File: lib/widgets/search/main_search_bar.dart [hunk 2]
```diff
@@ -110,16 +114,27 @@ class PositionedFloatingSearchBar extends SearchDelegate<String> {
 
   @override
   Widget buildSuggestions(BuildContext context) {
+    BuildingSingleton buildingSingleton = BuildingSingleton();
+    List<RoomCoordinate> roomList = <RoomCoordinate>[];
+    for (Building building in buildingSingleton.buildings) {
+      building.floors.forEach((floorName, floor) => floor
+          .coordinatesByGivenTypes({"ROOM"}).forEach(
+              (room) => roomList.add(room)));
+    }
```
Comment (ooms97) [inline lib/widgets/search/main_search_bar.dart]: I would move the logic of getting list of rooms to the building_singleton as this logic might be repeated many times in different widgets.

File: lib/widgets/search/main_search_bar.dart (hunk 1 above)
Comment (DiveshPatel) [reply lib/widgets/search/main_search_bar.dart]: Corrected!

File: lib/widgets/search/main_search_bar.dart (hunk 2 above)
Comment (DiveshPatel) [reply lib/widgets/search/main_search_bar.dart]: Done!

File: lib/data/building_singleton.dart [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import '../models/building.dart';
 import '../models/coordinate.dart';
+import '../models/coordinate.dart';
```
Comment (MrJCipherButtles) [inline lib/data/building_singleton.dart]: I forgot to mention this duplication in my last review

Comment (DiveshPatel) [reply lib/data/building_singleton.dart]: Corrected!

Comment (MrJCipherButtles) [reply lib/data/building_singleton.dart]: Seems like you fixed it as I sent my last comment 👍

**Pull request merged**